rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/responses/DropTableResponse'
+              example: { "dropped": true, "purged": false }
+        202:
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        404:
+          $ref: '#/components/schemas/ResponseErrorObject'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace.
+      responses:
+        200:
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+  /v1/tables/rename:
+    post:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name
+      description:
+        Rename a table from one identifier to another. It's valid to move a table
+        across namespaces, but the server implementation doesn't need to support it.
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        200:
+          description: OK
+        401:
+          description: Unauthorized
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, Table to rename does not exist
+            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceToRenameToDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          description: Not Acceptable - Unsupported Operation
+        409:
+          description: Conflict - The target table identifier to rename to already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              example:
+                $ref: '#/components/examples/TableAlreadyExistsError'
+
+components:
+  #######################################################
+  # Common Parameter Definitions Used In Several Routes #
+  #######################################################
+  parameters:
+    namespace:
+      name: namespace
+      in: path
+      required: true
+      description:
+        A namespace identifier as a single string.
+        Multipart namespace parts should be separated by the null byte, %00.
```

Comment:
I think we should remove `%00` here, but not from the example. The problem having it here is that we don't want people actually creating the string `accounting%00tax` and then URL encoding it to produce `accounting%2500tax`. Just saying `null byte` here is clear enough.

Reply:
What about from the example?

```
        multipart_namespace:
          value: "accounting%00tax"
```

Reply:
We spoke offline and we're keeping it in the examples to show. I might add a note bout the URL encoding later, once I find the right words for it.